JDStatusBarNotification/JDStatusBarNotification.m

```diff
@@ -289,10 +289,10 @@ - (void)dismissAnimated:(BOOL)animated;
   void(^complete)(BOOL) = ^(BOOL finished) {
     [self.overlayWindow removeFromSuperview];
     [self.overlayWindow setHidden:YES];
-    _overlayWindow.rootViewController = nil;
-    _overlayWindow = nil;
-    _progressView = nil;
-    _topBar = nil;
+    self.overlayWindow.rootViewController = nil;
+    self->_overlayWindow = nil;
+    self->_progressView = nil;
+    self->_topBar = nil;
```

Comment:
Fixing `JDStatusBarNotification.m:295:5: Block implicitly retains 'self';
explicitly mention 'self' to indicate this is intended behavior` warning. So I think self can be retaining until the animation finishes, right?

Reply:
Now that I'm looking at it more carefully, maybe a `cleanup` method would be nice.

```
- (void)cleanup {
    self.overlayWindow.rootViewController = nil;
    _overlayWindow = nil;
    _progressView = nil;
    _topBar = nil;
}
```

Reply:
yeah I agree, that would be even cleaner. Fell free to put it up!